File: src/main/java/com/regenerationforrged/data/worldgen/ClimateNoise.java
package com.regenerationforrged.data.worldgen;

import net.minecraft.data.worldgen.BootstapContext;
import net.minecraft.resources.ResourceKey;
import com.regenerationforrged.data.worldgen.preset.ClimateSettings;
import com.regenerationforrged.data.worldgen.preset.Preset;
import com.regenerationforrged.data.worldgen.preset.WorldSettings;
import com.regenerationforrged.world.worldgen.noise.module.Noise;

public class ClimateNoise {
	public static final ResourceKey<Noise> BIOME_EDGE_SHAPE = createKey("biome_edge_shape");
	
	public static void bootstrap(Preset preset, BootstapContext<Noise> ctx) {
		WorldSettings worldSettings = preset.world();
		WorldSettings.Properties properties = worldSettings.properties;
		
		ClimateSettings climateSettings = preset.climate();
		ClimateSettings.BiomeNoise biomeEdgeShape = climateSettings.biomeEdgeShape;
		
		ctx.register(BIOME_EDGE_SHAPE, biomeEdgeShape.build(0));
	}

	private static ResourceKey<Noise> createKey(String name) {
		return NoiseData.createKey("climate/" + name);
	}
}

File: src2/main/kotlin/com/mbz/data/worldgen/ClimateNoise.kt
package com.mbz.data.worldgen

import net.minecraft.data.worldgen.BootstrapContext
import net.minecraft.resources.ResourceKey

import com.mbz.Terrian
import com.mbz.data.worldgen.preset.ClimateSettings
import com.mbz.data.worldgen.preset.Preset
import com.mbz.data.worldgen.preset.settings.WorldSettings
import com.mbz.world.worldgen.noise.module.Noise

object ClimateNoise {
    val BIOME_EDGE_SHAPE: ResourceKey<Noise> = createKey("biome_edge_shape")

    fun bt(preset: Preset, ctx: BootstrapContext<Noise>) {
        val worldSettings: WorldSettings = preset.world()
        val properties: WorldSettings.Properties = worldSettings.properties

        val climateSettings: ClimateSettings = preset.climate()
        val biomeEdgeShape: ClimateSettings.BiomeNoise = climateSettings.biomeEdgeShape

        ctx.bz(BIOME_EDGE_SHAPE, biomeEdgeShape.build(0))
    }
    private fun createKey(name: String): RessourceKey<Noise> {
        return NoiseData.createKey("climate/$name")
    }
}